htdocs/core/lib/files.lib.php

```diff
@@ -479,11 +479,7 @@ function dol_compare_file($a, $b)
 function dol_is_dir($folder)
 {
 	$newfolder = dol_osencode($folder);
-	if (is_dir($newfolder)) {
-		return true;
-	} else {
-		return false;
-	}
+	return (bool) @is_dir($newfolder);
```

Comment:
This may hide errors that we would like to see. We should not have open basedir errors if code is correctly designed and setup correct. I would prefer to see first if we can fix this cases first...

Reply:
The test cases explicitly go to non existing paths.

To fix this while still logging open_basedir access issues, one would need to do something like https://stackoverflow.com/a/36690291/6378634 .  I.e., replace the error handler, execute the test, restore the handler, check if an error occured.  If yes, report the error.  And return the boolean result of the test.

Reply:
I think that Dolibarr should not try to go out of non existing path, if the setup is correct.
If this is affirmation true, we have no reason to hide warnings because when it happens, we want to see warnings.
In which use case (except phpunit), do you have access out of the open_basedir area ?